contracts/Parameterizer.sol

```diff
@@ -1,96 +1,323 @@
+  @notice for the provided proposal ID, set it, resolve its challenge, or delete it depending on whether it can be set, has a challenge which can be resolved, or if its "process by" date has passed
+  @param _propID the proposal ID to make a determination and state transition for
+  */
+  function processProposal(bytes32 _propID) public {
+    ParamProposal storage prop = proposalMap[_propID];
+
+    if (canBeSet(_propID)) {
+      set(prop.name, prop.value);
+    } else if (challengeCanBeResolved(_propID)) {
+      resolveChallenge(_propID);
+    } else if (now > prop.processBy) {
+    } else {
+      revert();
+    }
+
+    delete proposalMap[_propID];
+  }
+
+  /**
+  @notice claim the tokens owed for the msg.sender in the provided challenge
+  @param _challengeID the challenge ID to claim tokens for
+  @param _salt the salt used to vote in the challenge being withdrawn for
+  */
+  function claimReward(uint _challengeID, uint _salt) public {
+    // ensure voter has not already claimed tokens and challenge results have been processed
+    require(tokenClaims[_challengeID][msg.sender] == false);
+    require(challengeMap[_challengeID].resolved = true);
+
+    uint voterTokens = voting.getNumPassingTokens(msg.sender, _challengeID, _salt);
+    uint reward = calculateVoterReward(msg.sender, _challengeID, _salt);
+
+    // subtract voter's information to preserve the participation ratios of other voters
+    // compared to the remaining pool of rewards
+    challengeMap[_challengeID].totalTokens -= voterTokens;
+    challengeMap[_challengeID].rewardPool -= reward;
+
+    require(token.transfer(msg.sender, reward));
+    
+    // ensures a voter cannot claim tokens again
+    tokenClaims[_challengeID][msg.sender] = true;
+  }
+
+  // --------
+  // GETTERS:
+  // --------
+
+  /**
+  @dev Calculate the provided voter's token reward for the given poll
+  @param _voter Address of the voter whose reward balance is to be returned
+  @param _challengeID pollID of the challenge a reward balance is being queried for
+  @param _salt the salt for the voter's commit hash in the given poll
+  @return a uint indicating the voter's reward in nano-adToken
+  */
+  function calculateVoterReward(address _voter, uint _challengeID, uint _salt)
+  public constant returns (uint) {
+    uint totalTokens = challengeMap[_challengeID].totalTokens;
+    uint rewardPool = challengeMap[_challengeID].rewardPool;
+    uint voterTokens = voting.getNumPassingTokens(_voter, _challengeID, _salt);
+    return (voterTokens * rewardPool) / totalTokens;
+  }
+
+  /**
+  @notice Determines whether a proposal passed its application stage without a challenge
+  @param _propID The proposal ID for which to determine whether its application stage passed without a challenge
+  */
+  function canBeSet(bytes32 _propID) constant public returns (bool) {
+    ParamProposal memory prop = proposalMap[_propID];
+
+    return (propExists(_propID) && now > prop.appExpiry && now < prop.processBy && 
```

Comment:
Let's test this, but I think `propExists(_propID)` is extra because `now < prop.processBy` will effectively take care of this check.